test/e2e/backuppolicy/backuppolicy_test.go

```diff
@@ -0,0 +1,133 @@
+package backuppolicy
+
+import (
+	"fmt"
+	"time"
+
+	backupv1beta3 "github.com/anynines/a8s-backup-manager/api/v1beta3"
+	"github.com/anynines/a8s-deployment/test/framework"
+	policy "github.com/anynines/a8s-deployment/test/framework/backuppolicy"
+	"github.com/anynines/a8s-deployment/test/framework/dsi"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	replicas            = 1
+	suffixLength        = 5
+	pollingPeriod       = 1 * time.Second
+	asyncOpsTimeoutMins = time.Minute * 2
+)
+
+var (
+	backupPolicy *backupv1beta3.BackupPolicy
+	instance     dsi.Object
+)
+
+var _ = Describe("BackupPolicy", func() {
+	BeforeEach(func() {
+		// Create Dataservice instance and wait for instance readiness
+		instance, err = dsi.New(
+			dataservice,
+			testingNamespace,
+			framework.GenerateName(instanceNamePrefix, GinkgoParallelProcess(), suffixLength),
+			replicas,
+		)
+		Expect(err).To(BeNil(), "failed to generate new DSI resource")
+
+		Expect(k8sClient.Create(ctx, instance.GetClientObject())).
+			To(Succeed(), fmt.Sprintf("failed to create instance %s/%s",
+				instance.GetNamespace(), instance.GetName()))
+		dsi.WaitForReadiness(ctx, instance.GetClientObject(), k8sClient)
+	})
+
+	AfterEach(func() {
+
+		Expect(k8sClient.Delete(ctx, backupPolicy)).To(Succeed(),
+			fmt.Sprintf("failed to delete backup policy %s/%s",
+				backupPolicy.GetNamespace(), backupPolicy.GetName()))
+		Expect(k8sClient.Delete(ctx, instance.GetClientObject())).To(Succeed(),
+			fmt.Sprintf("failed to delete instance %s/%s",
+				instance.GetNamespace(), instance.GetName()))
+		Expect(policy.CleanupBackups(ctx, backupPolicy, k8sClient)).To(Succeed(), "failed to delete backups")
+		dsi.WaitForDeletion(ctx, instance.GetClientObject(), k8sClient)
+	})
+
+	Context("Backup Policy", func() {
+		It("Should Create BackupPolicy", func() {
+			By("Create a policy", func() {
+				backupPolicy = policy.New(
+					policy.SetNamespacedName(testingNamespace),
+					policy.SetEnabled(false),
+					policy.SetInstanceRef(instance),
+					policy.ScheduleConfiguration(backupv1beta3.ScheduleConfiguration{
+						Type:     backupv1beta3.ScheduleTypeCron,
+						TimeZone: "UTC",
+						Cron: backupv1beta3.Cron{
+							Expression: "2 2 * * *",
+						},
+					}))
+				Expect(k8sClient.Create(ctx, backupPolicy)).To(Succeed(),
```

Comment:
We don't just want to check that we pass API validation and can create the resource but also that it creates successfully via an expected status. Could we do something like that here?